api/envoy/api/v2/listener/listener.proto

```diff
@@ -113,14 +113,34 @@ message FilterChainMatch {
 
   // If non-empty, a transport protocol to consider when determining a filter chain match.
   // This value will be compared against the transport protocol of a new connection, when
-  // it's detected by one of the listener filters (this needs to be configured separately).
+  // it's detected by one of the listener filters.
   //
   // Valid values include:
   //
   // * ``raw_buffer`` - default, used when no transport protocol is detected,
   // * ``tls`` - set by :ref:`envoy.listener.tls_inspector <config_listener_filters_tls_inspector>`
   //   when TLS protocol is detected.
   string transport_protocol = 9;
+
+  // If non-empty, a list of application protocols to consider when determining a filter chain
+  // match. Those values will be compared against the application protocols of a new connection,
+  // when detected by one of the listener filters.
+  //
+  // Valid values include:
```

Comment:
small nit: Technically, if someone knows what they are doing and are doing custom clients, they can match on whatever they want, so I might switch "Valid" to "Suggested" or something like that. I don't feel that strongly about this, but the way it's written it seems like we will block values that aren't these which we aren't?

Reply:
Indeed, we're going to (ab)use this with `"istio"` for routing of the traffic that originated inside the service mesh :)

How about "Known" or "Popular", "Suggested" is fine, I guess... i.e.

    Known values include, but are not limited to:

Reply:
Updated both `transport_protocol` (since custom sniffers and transport socket implementations can support different values) and `application_protocols` to `Suggested`.